src/sentry/middleware/integrations/tasks.py

```diff
@@ -0,0 +1,142 @@
+            "slack.async_integration_response",
+            extra={
+                "path": webhook_payload.path,
+                "region": result["region"],
+                "region_status_code": result["response"].status_code,
+                "integration_status_code": integration_response.status_code,
+            },
+        )
+    except Exception as e:
+        sentry_sdk.capture_exception(e)
+
+
+@instrumented_task(
+    name="sentry.middleware.integrations.tasks.convert_to_async_discord_response",
+    queue="integrations.control",
+    silo_mode=SiloMode.CONTROL,
+)
+def convert_to_async_discord_response(
+    region_names: List[str],
+    payload: Dict[str, Any],
+    response_url: str,
+):
+    """
+    This task asks relevant region silos for response data to send asynchronously to Discord. It
+    assumes Discord has received a callback of type:5 (DEFERRED_CHANNEL_MESSAGE_WITH_SOURCE).
+    (See https://discord.com/developers/docs/interactions/receiving-and-responding#interaction-response-object-interaction-callback-type)
+
+    In the event this task finishes prior to returning the above type, the outbound post will fail.
+    """
+    webhook_payload = ControlOutbox.get_webhook_payload_from_outbox(payload=payload)
+    regions = [get_region_by_name(rn) for rn in region_names]
+    region_to_response_map: MutableMapping[str, Response] = {}
+    result: MutableMapping[str, Any] = {"response": None, "region": None}
+    for region in regions:
+        region_response = RegionSiloClient(region=region).request(
+            method=webhook_payload.method,
+            path=webhook_payload.path,
+            headers=webhook_payload.headers,
+            data=webhook_payload.body.encode("utf-8"),
+            json=False,
+            raw_response=True,
+        )
+        region_response = cast(Response, region_response)
+        region_to_response_map[region.name] = region_response
+        if region_response.status_code >= 200 and region_response.status_code < 300:
+            result["response"] = region_response
+            result["region"] = region
+
+    logger.info(
+        "discord.async_region_response",
+        extra={
+            "region": result["region"],
+            "response_map": {
+                region_name: response.status_code
+                for region_name, response in region_to_response_map.items()
+            },
+        },
+    )
+    if not result["response"]:
+        return
+
+    try:
+        # Region will return a response assuming it's meant to go directly to Discord. Since we're
+        # handling the request asynchronously, we extract only the data, and post it to the webhook
+        # that discord provides.
+        # https://discord.com/developers/docs/interactions/receiving-and-responding#followup-messages
+        payload = json.loads(result["response"].content.decode(encoding="utf-8")).get("data")
+        integration_response = requests.post(response_url, json=payload)
```

Comment:
@leeandher Discord will respond with 404 if this task completes before the control silo responds back to Discord. So we could retry the request again.